orchagent/cbforch.cpp

```diff
@@ -0,0 +1,322 @@
+                    erase(map_it);
+                }
+                else
+                {
+                    SWSS_LOG_ERROR("Failed to remove %s map %s",
+                                    getMapName(),
+                                    map_id.c_str());
+                }
+            }
+            else
+            {
+                SWSS_LOG_WARN("Tried to delete inexistent %s map %s",
+                               getMapName(),
+                               map_id.c_str());
+                // Mark it as success to remove it from the consumer
+                success = true;
+            }
+        }
+        else
+        {
+            SWSS_LOG_ERROR("Unknown operation type %s", op.c_str());
+            // Set success to true to remove the operation from the consumer
+            success = true;
+        }
+
+        if (success)
+        {
+            SWSS_LOG_DEBUG("Removing consumer item");
+            it = consumer.m_toSync.erase(it);
+        }
+        else
+        {
+            SWSS_LOG_DEBUG("Keeping consumer item");
+            ++it;
+        }
+    }
+}
+
+/*
+ * Purpose:     Extract the QoS map from the Redis's field-value pairs.
+ *
+ * Description: Iterate over the field-value pairs and create a SAI QoS map.
+ *
+ * Params:      IN  t - The field-value pairs.
+ *
+ * Returns:     The SAI QoS map object.
+ */
+sai_qos_map_list_t MapHandler::extractMap(const KeyOpFieldsValuesTuple &t)
+                                                                          const
+{
+    SWSS_LOG_ENTER();
+
+    sai_qos_map_list_t map_list;
+    const auto &fields_values = kfvFieldsValues(t);
+    map_list.count = (uint32_t)fields_values.size();
+    map_list.list = new sai_qos_map_t[map_list.count]();
+
+    uint32_t ind = 0;
+    for (auto i = fields_values.begin();
+         i != fields_values.end();
+         i++, ind++)
+    {
+        if (m_type == DSCP)
+        {
+            map_list.list[ind].key.dscp = (uint8_t)stoi(fvField(*i));
+        }
+        else
+        {
+            map_list.list[ind].key.mpls_exp = (uint8_t)stoi(fvField(*i));
+        }
+
+        map_list.list[ind].value.fc = (uint8_t)stoi(fvValue(*i));
+    }
+
+    return map_list;
+}
+
+/*
+ * Purpose:     Create a QoS map.
+ *
+ * Description: Create the QoS map over the SAI interface.
+ *
+ * Params:      IN  map_list - The QoS map to create.
+ *
+ * Returns:     The SAI ID for the newly created object or SAI_NULL_OBJECt_ID
+ *              if something goes wrong.
+ */
+sai_object_id_t MapHandler::createMap(const sai_qos_map_list_t &map_list)
+{
+    SWSS_LOG_ENTER();
+
+    vector<sai_attribute_t> map_attrs;
+
+    sai_attribute_t map_attr;
+    map_attr.id = SAI_QOS_MAP_ATTR_TYPE;
+    map_attr.value.u32 = m_type == DSCP ?
+                                SAI_QOS_MAP_TYPE_DSCP_TO_FORWARDING_CLASS :
+                                SAI_QOS_MAP_TYPE_MPLS_EXP_TO_FORWARDING_CLASS;
```

Comment:
Enclose ternary expressions in () brackets to make what's going on here a bit more readable.